### PR TITLE
Parser for WebKit's WebIDL dialect

### PR DESCRIPTION
**Pull request by mdittmer**
Also: Extra "parser consumed all input" check in existing parser integration tests.

### CODE REVIEW

File: test/any/parsing/WebKitParser-test.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  function parse(str, opt_production) {
+    var p = WebKitParser.create().parseString(str, opt_production);
+    expect(p.pos).toBe(str.length);
+    return p;
+  }
+
+  it('should parser example fragment from Location interface', function() {
+    var p = parse(
+      'SetterCallWith=ActiveWindow&FirstWindow',
```
Comment (arobins) [inline test/any/parsing/WebKitParser-test.js]: Maybe test that | works too?

File: lib/org/chromium/webidl/BaseParser.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      documentation: 'Grammar derived from "symbols", "actions", "separator".',
+      factory: function() {
+        var grammar = this.ImperativeGrammar.create({symbols: this.symbols});
+        grammar.addActions(this.actions);
+        return grammar;
+      }
+    },
+    {
+      name: 'ps',
+      documentation: 'Sting parser stream for synchronously parsing strings.',
```
Comment (arobins) [inline lib/org/chromium/webidl/BaseParser.js]: nit: string

File: lib/org/chromium/webidl/Parser.js [hunk 3]
```diff
@@ -72,749 +73,652 @@ foam.CLASS({
...
+              Typedef: tseq(
+                  'typedef', sym('Type'), sym('identifier'), sym('SemiColon')),
 
-          // Implements statements.
-          ImplementsStatement: tseq(
-            sym('identifier'), 'implements', sym('identifier'), sym('SemiColon')
-          ),
+              // Implements statements.
+              ImplementsStatement: tseq(
+                  sym('identifier'), 'implements', sym('identifier'),
```
Comment (arobins) [inline lib/org/chromium/webidl/Parser.js:232]: This all looks like formatting. I stopped reading.

File: lib/org/chromium/webidl/parsers/TokenParsers.js [hunk 4]
```diff
@@ -11,6 +11,9 @@ foam.CLASS({
   documentation: `An extension of foam.parse.Parsers with tokenized shorthands
     prefixed with "t".`,
 
+  // TODO(markdittmer): Implement complete parser description in
+  // someParser.toString(). Right now, all separator parsers
+  // of the same class share the same multion instance.
```
Comment (arobins) [inline lib/org/chromium/webidl/parsers/TokenParsers.js]: nit: multiton

**Review by arobins — APPROVED** [review]
lgtm

**Comment by mdittmer** [comment]
Towards #6

**Pull request merged**